src/assignments/tech-jobs-persistent.rst

```diff
@@ -0,0 +1,382 @@
+      method's parameters.
+
+#. Create a ``SkillController`` class and replicate the steps you followed above for ``EmployerController``.
+
+Test It!
+^^^^^^^^
+The employer and skill view templates for adding and viewing these objects are made for you. Before you move on,
+test your application now to make sure it runs as expected. You should be able to create Employer and Skill objects
+and view them.
+
+#. Start up your application – don’t forget to have our SQL server running – and go to the *Add Jobs*
+   view from the application's navigation menu.
+
+#. You won't be able to add a job yet, but you'll see a link to *Add Employers* and *Add Skills* in the form. Click them and proceed
+   to check the functionality of the forms that follow.
+
+#. Be sure to test your validation requirements and error handling.
+
+When everything works, move on to Part 2 below.
+
+.. admonition:: Tip
+
+   If everything seems to work – that is, you are
+   able to submit the form without any errors – but you don’t see your
+   employers or skills in the list after submission, here’s what you should check:
+
+   #. Is there any data in the ``employers`` and ``skills`` table? Check by going to MySQL Workbench
+      and looking for the employer/skill data within your schema.
+
+   #. If there’s data in the database, check that you are correctly
+      querying for the list of all objects in the controller
+      findAll()``?
+
+   #. Ensure you’re passing the list into the view, and it is named the same as the variable in the ThymeLeaf template.
+
+
+.. _tech-jobs-persistent-pt3:
+
+Part 3: Setting Up a One-to-Many Relationship
+---------------------------------------------
+
+In this application, any one ``Job`` object is affiliated with one employer while one ``Employer`` may contain several jobs.
+
+Now that you have set up persistence for the ``Employer`` and ``Skill`` classes, it is time to update the ``Job`` class
+to make use of these. ``Job`` is already using the Spring Data framework to be persistent and now you'll update its 
+``Employer`` field to create a one-to-many relationship. You'll also add a field on ``Employer`` to list the jobs associated 
+with each instance.
+
+Add a ``jobs`` Field to ``Employer``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Within ``Employer``, add a private property ``jobs`` of type
+   ``List<Job>`` and initialize it to an empty ``ArrayList``. After we
+   set up the ``Job`` class to work with ``Employer`` objects, this list
+   will represent the list of all items in a given job. We’ll do this
+   in a bit.
+
+#. Use the ``@OneToMany`` and ``@JoinColumn`` annotations on the jobs list in ``Employer`` to declare the relationship between   
+   data tables.
+
+Update ``Job`` Model
+^^^^^^^^^^^^^^^^^^^^
+
+#. Since it too has ``id`` and ``name`` fields, the ``Job`` model class can also inherit from ``AbstractEntity``. Update the 
```

Comment:
"Since the ``Job`` model class has ``id`` and ``name``...., it too can inherit from ``AbstractEntity``."